src/configurePipeline/configurePipelineSetup.ts

```diff
@@ -0,0 +1,128 @@
+		extensionVariables.pipelineTargetType = 1;
+		if (node instanceof WebAppTreeItem) {
+			extensionVariables.inputs.targetResource = extensionVariables.azureTargetService.extractTargetFromNode(node);
+		}
+		else {
+			extensionVariables.inputs.targetResource = await extensionVariables.azureTargetService.getTargetResource();
+		}
+
+		extensionVariables.azureDevOpsService = new AzureDevOpsService(extensionVariables.inputs.targetResource.credentials);
+		extensionVariables.sourceRepoService = new SourceRepoService();
+		extensionVariables.inputs.sourceRepoDetails = await extensionVariables.sourceRepoService.getSourceRepo(extensionVariables.azureDevOpsService);
+
+		extensionVariables.inputs.organizationName = await extensionVariables.azureDevOpsService.getOrganizationName();
+		extensionVariables.inputs.projectName = await extensionVariables.azureDevOpsService.getProjectName();
+
+		let fileUris = await vscode.window.withProgress({ location: vscode.ProgressLocation.Notification, title: "Analyzing your repo" }, analyzeRepo);
+		// Auto derieve this based on web app selected
+		var selectedPipeline = await getSelectedPipeline(fileUris[0], fileUris[1]);
+		extensionVariables.pipelineTargetType = getPipelineTargetType(selectedPipeline);
+
+		if (extensionVariables.inputs.sourceRepoDetails.sourceProvider === SourceProviderType.Github) {
+			let githubPat = await vscode.window.showInputBox({placeHolder: "Enter GitHub PAT token"})
+			extensionVariables.inputs.sourceProviderConnectionId = await extensionVariables.azureDevOpsService.createGitHubServiceConnection(githubPat);
+			extensionVariables.inputs.sourceProviderConnectionId = await extensionVariables.azureDevOpsService.getGitHubConnectionId();
+		}
+
+		extensionVariables.inputs.azureServiceConnectionId = await extensionVariables.azureDevOpsService.createAzureServiceConnection(extensionVariables.inputs)
+		let ymlFilePath: string = await extensionVariables.sourceRepoService.addYmlFileToRepo(getPipelineFilePath(selectedPipeline), extensionVariables.inputs);
+		await vscode.window.showTextDocument(vscode.Uri.file(ymlFilePath));
+		await vscode.window.showInformationMessage("Modify and commit yaml pipeline file to deploy.", "Commit", "Discard Pipeline")
+			.then((commitOrDiscard: string) => {
+				if (commitOrDiscard.toLowerCase() === "commit") {
+					return vscode.window.withProgress({ location: vscode.ProgressLocation.Notification, title: "Configuring Azure DevOps Pipeline and proceeding to deployment..." }, async (progress) => {
+						// handle when the branch is not upto date with remote branch and push fails
+						let commitDetails = await extensionVariables.sourceRepoService.commitAndPushPipelineFile(ymlFilePath);
+						extensionVariables.inputs.sourceRepoDetails.branch = commitDetails.branch;
+						extensionVariables.inputs.sourceRepoDetails.commitId = commitDetails.commitId;
+					});
+				}
+				else {
+					throw new Error("operation was discarded.");
```

Comment:
Do you want to throw error in this case ? Its a valid user input

Reply:
Will resolve this as part of stopping execution when users exits the configure pipeline setup at some point during the journey.

Reply:
Changed message to `Operation cancelled`. The same message is printed if the user quits the workflow at any point